src/aind_codeocean_utils/codeocean_job.py

```diff
@@ -0,0 +1,534 @@
+            update_data_perm_response = self.co_client.update_permissions(
+                data_asset_id=data_asset_id, everyone="viewer"
+            )
+            logger.info(
+                "Permissions response: "
+                f"{update_data_perm_response.status_code}"
+            )
+
+        return data_asset_reg_response
+
+    def _capture_result(
+        self,
+        computation_id: str,
+        input_data_asset_name: Optional[str],
+        capture_result_config: CaptureResultConfig,
+        additional_tags: Optional[list] = None,
+        additional_custom_metadata: Optional[dict] = None,
+    ) -> requests.Response:
+        """
+        Capture a result as a data asset. Can also share it with everyone.
+
+        Parameters
+        ----------
+        computation_id : str
+            ID of the computation to capture the result from.
+        input_data_asset_name : Optional[str]
+            Name of the input data asset to use to create the result name.
+        capture_result_config : CaptureResultConfig
+            The configuration for capturing the result, including:
+
+            * process_name : Optional[str]
+                Name of the process.
+            * mount : str
+                The mount folder name.
+            * asset_name : Optional[str]
+                The name to give the data asset.
+            * tags : Optional[List[str]]
+                The tags to use to describe the data asset.
+            * custom_metadata : Optional[dict]
+                What key:value metadata tags to apply to the asset.
+            * viewable_to_everyone : bool
+                Whether to share the captured results with everyone.
+        additional_tags : Optional[List[str]]
+            Additional tags to add to the data asset. If a data level RAW tag
+            is provided, it will be converted to DERIVED.
+        additional_custom_metadata : Optional[dict]
+            Additional custom metadata to add to the data asset. If a data
+            level custom metadata is provided, it will be converted to
+            DERIVED.
+
+        Returns
+        -------
+        requests.Response
+
+        """
+        if capture_result_config.asset_name is None:
+            assert (
+                input_data_asset_name is not None
+            ), "Either asset_name or input_data_asset_name must be provided"
+            capture_time = datetime_to_name_string(datetime.now())
+            asset_name = (
+                f"{input_data_asset_name}"
+                f"_{capture_result_config.process_name}"
+                f"_{capture_time}"
+            )
+        else:
+            asset_name = capture_result_config.asset_name
+
+        if capture_result_config.mount is None:
+            mount = capture_result_config.asset_name
+        else:
+            mount = capture_result_config.mount
+
+        tags = capture_result_config.tags.copy()
+        if additional_tags is not None:
+            tags += [
```

Comment:
The tags may be duplicated if they're in both capture_result_config.tags and additional_tags

Reply:
You're right! We should use sets maybe

Reply:
I'd be comfortable tackling this in a future pr

Reply:
Same

Reply:
done